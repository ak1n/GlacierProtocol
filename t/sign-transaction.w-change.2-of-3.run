#!/bin/bash
set -e

../../glacierscript.py --testnet=$1 sign-transaction << INPUT
y
y
y
y
y
y
y
2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
0200000000010154f3a5ba6a5a2beb6975c59895ad36fe84aa622edbaa662dadbd6e2a7a8a84670300000023220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3ffffffff0284434c000000000017a91422a07fe0ea8b8293eb336b9423f7e3958917924387c0e1e400000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac0400473044022078d174579919bc4fa7adb6813b5087a4bacd06014eb68a05232d728ec1ee3367022023583608c8f1a246cb61af5059d39cb66c4303e51ddff8789eb1fe4344788fc50100695221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae00000000
y
02000000015701865854493f0cb97b07ccf231003150433c74abc8cdac4c3c87fb25bbe9e0000000006a473044022003061e39e0eafff6120261e1930da298d14d46e594de1cf260cb7ef18446d3d3022010ff3990751a8e9cb90698223ca67607706a6d670ad9d1f63b55b560c73ab65a012102d69841fccc853bc99a1a32514d53d950528bd0eae03f45107cc10ce1ed4845acfeffffff05002d31010000000017a914fdd200f6e02076173292642fd352dc45f849070e8790409700000000001976a91414f909762e0f653521433c3d853d1f90dad17ee188ac002d31010000000017a91497c2ffdcdfc233a328751b46a47b781b1eec9b2d87002d31010000000017a91422a07fe0ea8b8293eb336b9423f7e3958917924387002d31010000000017a9142524a7e29329a636bf4c1d8dea0dc6a087e5d91687bd911300
1
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe
y
INPUT

# continuation of test create-withdrawal-data.partial-sign.w-change.2-of-3.run
# input data
#   extra "y" at beginning for confirming user aware only segwit support
#   then safety check "y"s as in create-withdrawal tests (6 "y")
#   cold storage address
#   partially signed tx hex = final output of create-withdrawal-data.2-of-3-segwit-partial-sign.golden
#   "y" to confirm data parsed from partially-signed tx hex
#   tx raw hex
#   1 = num keys to (re) sign with
#   key to sign with (2nd key from full/non-partial test)
#   "y" confirm re-sign setup
# output
#   should get "True" for sufficient private keys to execute
#   other data should match that in create-withdrawal
