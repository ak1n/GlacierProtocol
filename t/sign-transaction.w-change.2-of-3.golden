

Sequential signing of transactions supports ONLY Segwit wallets/transactions. This will not work with non-segwit wallets/transactions. Proceeding with non-segwit wallets/transactions RISKS LOSS OF FUNDS. Please confirm using segwit to proceed.
Confirm? (y/n): Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
For the partially-signed transaction

  Please paste the raw transaction (hexadecimal format) with unspent outputs at the source address
  OR
  input a filename located in the current directory which contains the raw transaction data
  (If the transaction data is over ~4000 characters long, you _must_ use a file.):

following variables parsed from partially signed hex input & storage address:

    cold storage / source_address: 2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N (for reference from manual input)

    redemption script: 5221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae

    destination address: mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99

    number of input transactions: 1

    change amount: 0.04998020

    withdrawal amount: 0.15000000


please confirm whether above data is correct before proceeding to input additional data for transaction re-sign
Confirm? (y/n): 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 btc (200.00000 mbtc)

How many private keys will you be signing this transaction with? 
#: Key #1: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 btc (200.00000 mbtc) in unspent supplied transactions
0.04998020 btc (49.98020 mbtc) going back to cold storage address 2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
0.15000000 btc (150.00000 mbtc) going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00001980 btc (0.01980 mbtc)

Signing with private keys: 
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
0200000000010154f3a5ba6a5a2beb6975c59895ad36fe84aa622edbaa662dadbd6e2a7a8a84670300000023220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3ffffffff0284434c000000000017a91422a07fe0ea8b8293eb336b9423f7e3958917924387c0e1e400000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac0400473044022053b2e6518e0846b5ff98b7f768a57484325c1f98e10b746f078a6f7a470a48280220172da2742182929bc9bd80eaeea8466c34984182fb59206da6326cb05f760abc01473044022078d174579919bc4fa7adb6813b5087a4bacd06014eb68a05232d728ec1ee3367022023583608c8f1a246cb61af5059d39cb66c4303e51ddff8789eb1fe4344788fc501695221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae00000000

Transaction fingerprint (md5):
0b11452b57a52e9c0e617a89d9d3984b
QR code for transaction written to transaction.png
